daemon/config/config_linux.go

```diff
@@ -136,6 +145,16 @@ func (conf *Config) LookupInitPath() (string, error) {
 	return exec.LookPath(binary)
```

Comment:
This fallback means we will look in `$PATH`, so this would also include, e.g. `docker-compose` 🤔 😂 

```bash
cp /usr/libexec/docker/cli-plugins/docker-compose /usr/local/bin
dockerd --init-path=docker-compose
...
WARN[2024-05-10T13:16:18.581826096Z] failed to parse /usr/local/bin/docker-compose version  error="unknown output format: Docker Compose version v2.25.0\n"


docker info | grep Init
 Init Binary: docker-hello

docker run -dit --init busybox
dac20108102f70578ed24d0ba5952bb874130d14882952e37dfbfb8bc19b4e5a

docker ps -a
CONTAINER ID   IMAGE     COMMAND   CREATED          STATUS                        PORTS     NAMES
dac20108102f   busybox   "sh"      59 seconds ago   Exited (16) 58 seconds ago              relaxed_boyd
```

Reply:
Sure, but they could also have done `--init-path=/usr/libexec/docker/cli-plugins/docker-compose`.
I'm not understanding the concern here. We aren't resolving the wrong binaries except that the user potentially put in the wrong binary.